binscripts/rvm-installer

```diff
@@ -838,20 +843,23 @@ rvm_install_select_and_get_version()
   case "${version}" in
     (head)
       _version_release="${branch}"
-      install_head sources[@] ${branch:-master} || exit $?
+      install_head sources[@] ${branch:-master} || fail_with_code $? "Failed to install version '$version'"
       ;;
 
     (latest)
-      install_release sources[@] $(fetch_version sources[@]) || exit $?
+      _version=$(fetch_version sources[@])   || fail "Failed to fetch version '$version'"
+      install_release sources[@] "$_version" || fail_with_code $? "Failed to install version '$version'"
       ;;
 
     (latest-minor)
       version="$(\cat "$rvm_path/VERSION")"
-      install_release sources[@] $(fetch_version sources[@] ${version%.*}) || exit $?
+      _version=$(fetch_version sources[@] ${version%.*}) || fail "Failed to fetch version '$version'"
+      install_release sources[@] "$_version"             || fail_with_code $? "Failed to install version '$version'"
       ;;
 
     (latest-*)
-      install_release sources[@] $(fetch_version sources[@] ${version#latest-}) || exit $?
+      _version=$(fetch_version sources[@] ${version#latest-}) || fail "Failed to fetch version '$version'"
+      install_release sources[@] "$_version"                  || fail_with_code $? "Failed to install version '$version'"
```

Comment:
I broke these `install_release ... $(fetch_version ...)` statements into 2 lines here, so that we can exit (fail_with_code) immediately after returning from `fetch_version`.

Alternatively, I could have just `exit`ed directly within `fetch_version` on line 263 when it detects failure.  But I am always hesitant to add `exit` to additional functions, as it adds complexity.

Still, I'm open to changing this to exit early in fetch_version if it helps reduce the delta.